### PR TITLE
Revamp webhook documentation

### PR DESCRIPTION
**Pull request by madtrick**
### Summary

With the release of the new webhooks infrastructure we have to also update our documentation to include all the new features (and also make it more understandable). Main changes:

- Describe what are topics. Explain its syntax and the `*` character.
- Describe the `auto_save` event
- Fix some errors in the request/response descriptions

### CODE REVIEW

File: apiary/cma.apib [hunk 1]
```diff
@@ -855,75 +855,84 @@ To use search parameters, these should be put in the end of your query, similar
...
+These restrictions have to be translated into a list of `[Action].[Type]` pairs, `[*.publish, Asset.delete, Entry.*]`, and included in the payload that defines the webhook.
 
-For example, when an Entry has been published:
+`*` is a wildcard character that can be used in place of any Action or Type. The combination `*.*` is also valid and it means that your webhook will be subscribed to all actions across all types.  It's important that you keep in mind that using `*` opens the door for your webhook to be called for future actions or types that didn't exist when the webhook was created or updated.
+#### Auto save
 
-```
-X-Contentful-Topic: ContentManagement.Entry.publish
-```
+The Contentful Webapp automatically saves the documents you are working on after every change you make. Besides that, Contentful keeps track of those documents that are active in the Webapp and uses that information to call configured webhooks. For this matter Contentful considers active those documents had been edited in the last 20 seconds.
```
Comment (TimBeyer) [inline apiary/cma.apib]: > For this matter Contentful considers active those documents had been edited in the last 20 seconds.
I think there's a 'that' missing here.

File: apiary/cma.apib [hunk 2]
```diff
@@ -855,75 +855,84 @@ To use search parameters, these should be put in the end of your query, similar
...
+
+This means that, for example, if you are editing an Entry in the Contentful Webapp for one minute, and you have a webhook configured to be called for `auto_save` actions, that webhook will be called 3 trimes.
```
Comment (TimBeyer) [inline apiary/cma.apib]: > called 3 trimes.

'3 times'

File: apiary/cma.apib [hunk 3]
```diff
@@ -855,75 +855,84 @@ To use search parameters, these should be put in the end of your query, similar
...
+Let's assume that:
 
-```json
-{
-  "name": "My webhook",
-  "url": "https://www.example.com",
-  "headers": [{ "key": "X-My-Custom-Header", "value": "some value"}]
-}
-```
+- One of your entries have just been published
```
Comment (TimBeyer) [inline apiary/cma.apib]: 'has just been published'

File: apiary/cma.apib [hunk 4]
```diff
@@ -855,75 +855,84 @@ To use search parameters, these should be put in the end of your query, similar
...
-### Body of Webhook HTTP Request
+### Body
 
-The request body is the published Resource's body or deletion thereof.
+The payload received by a webhook will change depending on the action type:
 
-The request body's content depends on the specific action:
+- For `Publish` it will receive the latest published version. That's the same payload as the one available through the Delivery API with the exception that the payload for Entries and Assets will contain all locales, e.g. instead of `fields.name` there will be `fields['en-US'].name`.
+- For `Unpublish` and `Delete` it will receive a deletion object. In the case of Entries the Deletions are extended with the id of their Content Type.
+- For `Auto Save`, `Archive`, `Unarchive` and `Create` it will receive the latest draft version at the time the webhook was triggered. That's the same payload as the one available throgh the Management API.
```
Comment (TimBeyer) [inline apiary/cma.apib]: 'through'

File: apiary/cma.apib [hunk 5]
```diff
@@ -964,13 +973,12 @@ Or a Deletion of an Entry when an Entry has been unpublished:
 }
 ```
 
-### Response Detection
+### Error handling
 
-When a webhook is triggered an HTTP POST request is made to its pre-configured HTTP URL.
-Depending on the HTTP response's status code the webhook will either succeed or fail:
+Contentful will use the HTTP response status code of the webhook call to decide wether it succeeded or failed:
```
Comment (TimBeyer) [inline apiary/cma.apib]: 'whether'

File: apiary/cma.apib [hunk 6]
```diff
@@ -1000,8 +1008,11 @@ increasing delays in between up to a maximum of 3 attempts.
             Content-Type: application/vnd.contentful.management.v1+json
 
     + Attributes
```
Comment (TimBeyer) [inline apiary/cma.apib]: This appears a few times. Shouldn't it be possible to extract this into Data Structures (https://github.com/contentful/slash-developers/blob/master/apiary/cma.apib#L1413)?

Comment (TimBeyer) [inline apiary/cma.apib]: Ah I guess some of them have slight differences.

File: apiary/cma.apib (hunk 1 above)
Comment (madtrick) [inline apiary/cma.apib]: fixed

File: apiary/cma.apib (hunk 2 above)
Comment (madtrick) [inline apiary/cma.apib]: fixed

File: apiary/cma.apib (hunk 3 above)
Comment (madtrick) [inline apiary/cma.apib]: fixed

File: apiary/cma.apib (hunk 6 above)
Comment (madtrick) [inline apiary/cma.apib]: hm, let me take a look

File: apiary/cma.apib (hunk 4 above)
Comment (madtrick) [inline apiary/cma.apib]: fixed

File: apiary/cma.apib (hunk 5 above)
Comment (madtrick) [inline apiary/cma.apib]: fixed

File: apiary/cma.apib (hunk 6 above)
Comment (madtrick) [inline apiary/cma.apib]: Extracted the common attributes https://github.com/contentful/slash-developers/pull/254/commits/941450806717abba1f87f50e23bb19de5fc3ab61#diff-c14df00ddf70646286212a701d82e78eR1643

File: apiary/cma.apib [hunk 7]
```diff
@@ -855,75 +855,84 @@ To use search parameters, these should be put in the end of your query, similar
...
-For example, when an Entry has been published:
+`*` is a wildcard character that can be used in place of any Action or a Type. The combination `*.*` is also valid and it means that your webhook will be subscribed to all actions across all types.  It's important that you keep in mind that using `*` opens the door for your webhook to be called for new actions or types that didn't exist when the webhook was created or updated.
+#### Auto save
 
-```
-X-Contentful-Topic: ContentManagement.Entry.publish
-```
+The Contentful Webapp saves the documents you are working on automatically after every change you make. Besides that, Contentful keeps track of those documents that are active in the Webapp and uses that information to call configured webhooks. For this matter Contentful considers active those documents had been edited in the last 20 seconds.
+
+This means that, for example, if you are editing an Entry in the Contentful Webapp for one minute, and you have a webhook configured to be called for `auto_save` actions, that webhook will be called 3 trimes.
```
Comment (eugenekudashev) [inline apiary/cma.apib]: 3 times

File: apiary/cma.apib [hunk 8]
```diff
@@ -855,75 +849,84 @@ To use search parameters, these should be put in the end of your query, similar
...
+These restrictions have to be translated into a list of `[Action].[Type]` pairs, `[*.publish, Asset.delete, Entry.*]`, and included in the payload that defines the webhook.
 
-For example, when an Entry has been published:
+`*` is a wildcard character that can be used in place of any Action or Type. The combination `*.*` is also valid and it means that your webhook will be subscribed to all actions across all types.  It's important that you keep in mind that using `*` opens the door for your webhook to be called for future actions or types that didn't exist when the webhook was created or updated.
+#### Auto save
 
-```
-X-Contentful-Topic: ContentManagement.Entry.publish
-```
+The Contentful Webapp automatically saves the documents you are working on after every change you make. Besides that, Contentful keeps track of those documents that are active in the Webapp and uses that information to call configured webhooks. For this matter Contentful considers active those documents that had been edited in the last 20 seconds.
```
Comment (eugenekudashev) [inline apiary/cma.apib]: Web App

File: apiary/cma.apib [hunk 9]
```diff
@@ -855,75 +849,84 @@ To use search parameters, these should be put in the end of your query, similar
...
-  "url": "https://www.example.com",
-  "headers": [{ "key": "X-My-Custom-Header", "value": "some value"}]
-}
-```
+- One of your entries has just been published.
+- The name of your webhook is `Very hook`.
+- You have defined two custom headers: `X-Very: wow` and `Authentication: indeed`.
 
-If someone is now publishing an entry, the following headers will be sent:
+On this scenario your webhook would we called with the following set of headers:
```
Comment (eugenekudashev) [inline apiary/cma.apib]: **In** this scenario

File: apiary/cma.apib (hunk 8 above)
Comment (madtrick) [inline apiary/cma.apib]: Fixed

File: apiary/cma.apib (hunk 9 above)
Comment (madtrick) [inline apiary/cma.apib]: Fixed

**Pull request merged**